File: online-judge-solutions/Codeforces/373A.py
#https://codeforces.com/problemset/problem/373/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
k=int(input())
m=[int(c) for row in range(4) for c in input().strip() if c!='.']
print("YES" if all(m.count(t) <= 2*k for t in set(m)) else "NO")

File: online-judge-solutions/Codeforces/373A.cpp
//https://codeforces.com/problemset/problem/373/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
#include<map>
using namespace std;
int main(){
	int k;
	while(cin>>k){
		int n=16;
		char x;
		bool b=true;
		map<int,int> m;
		while(n--){
			cin>>x;
			if(x-'0'>=0 && x-'0'<=9) m[x-'0']++;
			if(m[x-'0']>(2*k)) b=false;
		}
		cout<<(b?"YES":"NO")<<"\n";
	}
	return 0;
}
